analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.PatternMatching.cs

```diff
@@ -198,6 +198,23 @@ public void RecursivePattern_SetsNotNull_Nested()
             validator.TagValues("End").Should().HaveCount(2).And.OnlyContain(x => x != null && x.HasConstraint(TestConstraint.First));   // 2x because value has different states
         }
 
+        [TestMethod]
+        public void ResursivePattern_ExistingConstraint_DoesNothing()
+        {
+            const string code = @"
+object value = arg switch
+{
+    null => null,
+    { } when Condition => null, // Should not create arg=Null
+    _ => Tag(""Arg"", arg)
+};
+
+static object Tag(string name, object value) => null;";
```

Comment:
Why do we need this helper here? Isn't a suitable "Tag" method in scope already?